synapse/storage/e2e_room_keys.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 New Vector Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from twisted.internet import defer
+
+from ._base import SQLBaseStore
+
+
+class EndToEndRoomKeyStore(SQLBaseStore):
+
+    @defer.inlineCallbacks
+    def get_e2e_room_key(self, user_id, version, room_id, session_id):
+
+        row = yield self._simple_select_one(
+            table="e2e_room_keys",
+            keyvalues={
+                "user_id": user_id,
+                "version": version,
+                "room_id": room_id,
+                "session_id": session_id,
+            },
+            retcols=(
+                "first_message_index",
+                "forwarded_count",
+                "is_verified",
+                "session_data",
+            ),
+            desc="get_e2e_room_key",
+        )
+
+        defer.returnValue(row)
+
+    def set_e2e_room_key(self, user_id, version, room_id, session_id, room_key):
+
+        def _set_e2e_room_key_txn(txn):
+
+            self._simple_upsert_txn(
+                txn,
+                table="e2e_room_keys",
+                keyvalues={
+                    "user_id": user_id,
+                    "room_id": room_id,
+                    "session_id": session_id,
+                },
+                values={
+                    "version": version,
+                    "first_message_index": room_key['first_message_index'],
+                    "forwarded_count": room_key['forwarded_count'],
+                    "is_verified": room_key['is_verified'],
+                    "session_data": room_key['session_data'],
+                },
+                lock=False,
+            )
+
+            return True
+
+        return self.runInteraction(
+            "set_e2e_room_key", _set_e2e_room_key_txn
+        )
+
+    # XXX: this isn't currently used and isn't tested anywhere
+    # it could be used in future for bulk-uploading new versions of room_keys
+    # for a user or something though.
```

Comment:
Unless you're planning on implementing this I'd rather we just got rid of it, especially if it isn't tested.

Reply:
i'd rather not lose the code on the floor, given it may well be useful in future.

Reply:
I'd rather not have to review code that isn't used, and dead code just adds bloat to the code base and makes understanding things more annoying.

Also, I think I would ask you to rewrite.

Reply:
ok, done